### PR TITLE
feat: ruby local evaluation docs

### PR DESCRIPTION
**Pull request by yuhao900914**
# Dev Docs PR

## Description

Describe your changes. 

## Deadline

When do these changes need to be live on the site?

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [x] Doc update.
- [x] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [ ] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [ ] Running `mkdocs serve` didn't generate any failures.
- [ ] I have performed a self-review of my own documentation.

@amplitude-dev-docs
@caseyamp

### CODE REVIEW

**Review by caseyamp — APPROVED** [review]
LGTM from a language standpoint, recommend having a peer review for technical accuracy.

File: docs/experiment/sdks/ruby-sdk.md [hunk 1]
```diff
@@ -47,7 +50,7 @@ Install the Ruby Server SDK with bundler or gem directly.
     require 'amplitude-experiment'
 
     # (1) Initialize the experiment client
-    experiment = AmplitudeExperiment.init('<DEPLOYMENT_KEY>', AmplitudeExperiment::Config.new)
+    experiment = AmplitudeExperiment.init_remote('<DEPLOYMENT_KEY>', AmplitudeExperiment::Config.new)
```
Comment (qingzhuozhen) [inline docs/experiment/sdks/ruby-sdk.md:53]: The config name?

File: docs/experiment/sdks/ruby-sdk.md [hunk 2]
```diff
@@ -168,6 +171,156 @@ experiment.fetch_async(user) do |_, variants|
...
+
+!!!warning "Local Evaluation Mode"
+    The local evaluation client can only evaluation flags which are [set to local evaluation mode](../guides/create-local-evaluation-flag.md).
+
+### Install
+
+Install the Ruby Server SDK's local evaluation.
+
+!!!warning "OS, and architecture support"
+    Local evaluation requires `CGO` be enabled (`CGO_ENABLED=1`). Additionally, the local evaluation package currently only supports the following OS' and architectures (`GOOS/GOARCH`):
```
Comment (qingzhuozhen) [inline docs/experiment/sdks/ruby-sdk.md:186]: cgo?

File: docs/experiment/sdks/ruby-sdk.md [hunk 3]
```diff
@@ -168,6 +171,156 @@ experiment.fetch_async(user) do |_, variants|
...
+
+    1. [Initialize the experiment client](#initialize)
+    2. [Start the local evaluation client](#start)
+    3. [Evaluate a user](#evaluate)
+
+    ```ruby
+    require 'amplitude-experiment'
+
+    # (1) Initialize the local evaluation client with a server deployment key.
+    experiment = AmplitudeExperiment.init_remote('<DEPLOYMENT_KEY>', AmplitudeExperiment::Config.new)
```
Comment (qingzhuozhen) [inline docs/experiment/sdks/ruby-sdk.md:227]: init_local?

File: docs/experiment/sdks/ruby-sdk.md [hunk 4]
```diff
@@ -168,6 +171,156 @@ experiment.fetch_async(user) do |_, variants|
...
+
+    # (2) Fetch variants for a user
+    user = AmplitudeExperiment::User.new(
+      user_id: 'user@company.com',
+      device_id: 'abcdefg',
+      user_properties: {
+        'premium' => true
+      }
+    )
+    variants = experiment.fetch(user)
```
Comment (qingzhuozhen) [inline docs/experiment/sdks/ruby-sdk.md:237]: example

File: docs/experiment/sdks/ruby-sdk.md [hunk 5]
```diff
@@ -168,6 +171,156 @@ experiment.fetch_async(user) do |_, variants|
...
+AmplitudeExperiment.init_local(api_key)
+```
+
+| Parameter | Requirement | Description |
+| --- | --- | --- |
+| `apiKey` | required | The server [deployment key](../general/data-model.md#deployments) which authorizes fetch requests and determines which flags should be evaluated for the user. |
+| `config` | optional | The client [configuration](#configuration) used to customize SDK client behavior. |
+
+!!!tip "Flag Polling Interval"
+    Use the `flag_config_polling_interval_millis` [configuration](#configuration_1) to determine the time flag configs take to update once modified (default 30s).
```
Comment (qingzhuozhen) [inline docs/experiment/sdks/ruby-sdk.md:267]: is that `configuration-1`?

File: docs/experiment/sdks/ruby-sdk.md [hunk 6]
```diff
@@ -168,6 +171,156 @@ experiment.fetch_async(user) do |_, variants|
...
+
+The SDK client can be configured on initialization.
+
+???config "Configuration Options"
+    | <div class="big-column">Name</div> | Description | Default Value |
+    | --- | --- | --- |
+    | `debug` | Set to `true` to enable debug logging. | `false` |
+    | `server_url` | The host to fetch flag configurations from. | `https://api.lab.amplitude.com` |
+    | `flag_config_polling_interval_millis` | The interval to poll for updated flag configs after calling [`start`](#start) | `30000` |
+    | `flag_config_poller_request_timeout_millis` | The timeout for the request made by the flag config poller | `10000` |
```
Comment (qingzhuozhen) [inline docs/experiment/sdks/ruby-sdk.md:279]: Do you have this in ruby?

File: docs/experiment/sdks/ruby-sdk.md [hunk 7]
```diff
@@ -168,6 +171,156 @@ experiment.fetch_async(user) do |_, variants|
...
+!!!warning "Server Deployment Key"
+    You must [initialize](#initialize_1) the local evaluation client with a server [deployment](../general/data-model.md#deployments) key in order to get access to local evaluation flag configs.
+
+```ruby
+AmplitudeExperiment.init_local(api_key)
+```
+
+| Parameter | Requirement | Description |
+| --- | --- | --- |
+| `apiKey` | required | The server [deployment key](../general/data-model.md#deployments) which authorizes fetch requests and determines which flags should be evaluated for the user. |
```
Comment (qingzhuozhen) [inline docs/experiment/sdks/ruby-sdk.md:263]: nit: the api key cases could be aligned

File: docs/experiment/sdks/ruby-sdk.md (hunk 2 above)
Comment (yuhao900914) [reply docs/experiment/sdks/ruby-sdk.md:186]: I saw python also includes the `cgo` description. Should that also be removed?

Comment (qingzhuozhen) [reply docs/experiment/sdks/ruby-sdk.md:186]: Good catch, I removed from the title and forgot in the text, I will create a PR to clean this up

File: docs/experiment/sdks/ruby-sdk.md (hunk 5 above)
Comment (yuhao900914) [reply docs/experiment/sdks/ruby-sdk.md:267]: It should be `#configuration_1`

File: docs/experiment/sdks/python-sdk.md [hunk 8]
```diff
@@ -252,7 +252,7 @@ The SDK client can be configured on initialization.
 
 ### Start
 
-Start the local evaluation client, pre-fetching local evaluation mode flag configs for [evaluation](#evaluate) and starting the flag config poller at the [configured](#configuration_1) interval.
+Start the local evaluation client, pre-fetching local evaluation mode flag configs for [evaluation](#evaluate) and starting the flag config poller at the [configured](#configuration) interval.
```
[the remark this answers is not in the document]
Comment (yuhao900914) [reply docs/experiment/sdks/python-sdk.md:255]: Wrong edition.

**Review by qingzhuozhen — APPROVED** [review]
lg, may need to hold till the release go out

File: docs/experiment/sdks/ruby-sdk.md [hunk 9]
```diff
@@ -168,6 +171,157 @@ experiment.fetch_async(user) do |_, variants|
...
+    | `flag_config_polling_interval_millis` | The interval to poll for updated flag configs after calling [`start`](#start) | `30000` |
+    | `debug` | Set to `true` to enable debug logging. | `false` |
+
+### Start
+
+Start the local evaluation client, pre-fetching local evaluation mode flag configs for [evaluation](#evaluate) and starting the flag config poller at the [configured](#configuration_1) interval.
+
+```ruby
+start
+```
```
Comment (bgiori) [inline docs/experiment/sdks/ruby-sdk.md:288]: is this how you call the function on the client? Or is this the function "definition"

**Review by bgiori — APPROVED** [review]
🎉

File: docs/experiment/sdks/ruby-sdk.md (hunk 9 above)
Comment (yuhao900914) [reply docs/experiment/sdks/ruby-sdk.md:288]: I think it's the function "definition".  It's followed by how we call on clients.
I copied from python, 
Screen Shot 2022-08-22 at 10 30 16 PM
This is how python shows now.

**Pull request merged**